cstrml/market/src/lib.rs

```diff
@@ -172,6 +186,10 @@ decl_storage! {
         /// Order details iterated by order id
         pub StorageOrders get(fn storage_orders):
         map hasher(twox_64_concat) T::Hash => Option<StorageOrder<T::AccountId, BalanceOf<T>>>;
+
+        /// Pledge details iterated by provider id
+        pub PledgeLedgers get(fn pledge_ledgers):
+        map hasher(twox_64_concat) T::AccountId => Option<PledgeLedger<BalanceOf<T>>>;
```

Comment:
As for `PledgeLedger`, I don't think it need `Option`, cause `total = 0` means `None` already, remove `Option` can simplify the code logic